### PR TITLE
Implement basic Pick Up Winner Flow

### PR DESCRIPTION
**Pull request by aEricLiao**
- Implement basic Pick Up Winner Flow

### CODE REVIEW

File: ui/schema.graphql [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+
+#for temp
+type RootMutationType {
+  findWinner(awardId: String): Candidate
+}
+
+#for temp
+type Candidate {
+  id: ID!
+  name: String!
```
Comment (aStanleyLiang) [inline ui/schema.graphql:14]: nice, you start to graphQL as well 🎉

File: ui/src/containers/Drawer/Drawer.tsx [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 import { useHistory } from 'react-router-dom'
 import Konva from 'konva'
 import { IFrame } from 'konva/types/types'
 import { Layer } from 'konva/types/Layer'
 import Wheel from './components/Wheel'
+import { winnerVar } from '@src/config/apolloClient'
+import { CUR_AWARD_QUERY } from '@src/query/award'
+import { CHOOSE_WINNER } from '@src/query/winner'
+import { GetCurrentAward } from '@src/__generated__/GetCurrentAward'
+import { ChooseWinner } from '@src/__generated__/ChooseWinner'
```
Comment (aStanleyLiang) [inline ui/src/containers/Drawer/Drawer.tsx:13]: I noticed the folder name '__generated__'? Is it created by graphQL?

File: ui/src/containers/Winner/index.tsx [hunk 3]
```diff
@@ -22,12 +28,12 @@ const Winner = ({}) => {
         <img className={style.winner_gif_container} src={gif} alt='' />
         <div className={style.result}>
           <div className={style.result_award}>
-            <span>$10,000</span>
+            <div>{priceText && (priceText.length > 20 ? priceText.slice(0, 20) + '...' : priceText)}</div>
```
Comment (aStanleyLiang) [inline ui/src/containers/Winner/index.tsx]: I suggest to use css line clamp to replace this, check the article here

https://css-tricks.com/almanac/properties/l/line-clamp/

**Review by aStanleyLiang — CHANGES_REQUESTED** [review]
Well done. I suggest to use css line-clamp to replace the dot dot dot case

**Comment by aMichaelPoernomo** [comment]
Some suggestion~
I think it's better to have `winner` field with `Candidate` return type in `Award` query in schema `luckydraw/ui/schema.graphql`, so we can remove unused `winner` query.
CMIIW, but I think returning `Candidate` won't trigger changes in apollo cache for both `awards` and `curAward` query, since it mutates `Candidate` value not `Award` value.

**Review by aMichaelPoernomo — APPROVED** [review]

File: ui/src/containers/Drawer/Drawer.tsx (hunk 2 above)
Comment (aEricLiao) [reply ui/src/containers/Drawer/Drawer.tsx:13]: Yes, it created by` apollo-tooling`  to gen the typescript definition for query.
https://www.npmjs.com/package/apollo#apollo-clientcodegen-output

strange that the GitHub page remove this command now @@,I saw it few day before , maybe it's changing  sth

File: ui/src/containers/Winner/index.tsx (hunk 3 above)
Comment (aEricLiao) [reply ui/src/containers/Winner/index.tsx]: is it need to add with`-webkit-` prefix ,and changing the display attribute?
https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp

**Comment by aEricLiao** [comment]
> Some suggestion~
> I think it's better to have `winner` field with `Candidate` return type in `Award` query in schema `luckydraw/ui/schema.graphql`, so we can remove unused `winner` query.
> CMIIW, but I think returning `Candidate` won't trigger changes in apollo cache for both `awards` and `curAward` query, since it mutates `Candidate` value not `Award` value.

Yep~It would be more simple that we just using the `winner` in the `awards` query, and using [ fetch-policy](https://www.apollographql.com/docs/react/data/queries/#setting-a-fetch-policy) to  ensure fetching the newest data. I just wanna try using  the local state management of Apollo  😅

**Review by aYishinHo — APPROVED** [review]
kamidesu!

**Comment by aStanleyLiang** [comment]
Sorry that I didn't approve that by far I fail to docker-compose on my side. I will seek for help today

**Review by aStanleyLiang — CHANGES_REQUESTED** [review]
I noticed a condition that drawer page is not scroll to top after I start to draw from the award list. To fix this a window.scrollTop = 0 is required in drawer.tsx

**Comment by aEricLiao** [comment]
fix in https://github.com/aStanleyLiang/luckydraw/pull/17/commits/6e1037db7d34313665a10b9d0c41bfe2783b19e5 thx

**Review by aStanleyLiang — APPROVED** [review]
good job

**Comment by aStanleyLiang** [comment]
I directly merge this to check the latest status

**Pull request merged**